okapi-common/src/main/java/org/folio/okapi/common/refreshtoken/tokencache/impl/TokenCacheImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.okapi.common.refreshtoken.tokencache.impl;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.folio.okapi.common.refreshtoken.tokencache.TokenCache;
+
+public class TokenCacheImpl implements TokenCache {
+
+  static class CacheValue {
+    String value;
+    long expires;
+
+    boolean expired() {
+      return expires < System.currentTimeMillis();
+    }
+  }
+
+  final Map<String,CacheValue> entries;
```

Comment:
I'm not understanding the problem that having the map or the cache solves. Why not just have a reference to the token that is bound to the credentials of the client? There's only one AT or RT that belongs to given credentials isn't there? And if that token has gone bad, doesn't this code wait until it gets a new one in `getTokenWithExpiry` either way?

Reply:
We need a map because we want to avoid logging in all the time. We have multiple tenants, users and passwords. And the two former serves as key. This is optimization only.

Reply:
I see. I didn't notice that the token cache is passed into the constructor of the client. Makes sense now.